src/project_euler/_3/mod.rs

```diff
@@ -0,0 +1,36 @@
+//! Larget Prime Factor
+//!
+//! Problem #3.
+//! https://projecteuler.net/problem=3
+
+/// Get the largest prime factor of the (unsigned) integer `n`
+pub fn largest_prime_factor(mut n: u64) -> u64 {
+    // Largest prime factor
+    let mut lpf = 2;
+    while n > lpf {
+        if n % lpf == 0 {
+            n /= lpf;
+            lpf = 2;
```

Comment:
why do we have `lpf=2` here?

Reply:
Because if we don't reset it, the loop would never terminate (try it!). Notice the while loop condition. The way this algorithm works is by dividing `n` by primes each time (`lpf` is always prime, unless it's `1`, because `2` is a prime, and if it can ever divide perfectly into `n`), untill `n` is equal to or less than `lpf`.